### PR TITLE
BE-1108 update test to remove Windows-only code, update readme & delays

### PR DESCRIPTION
**Pull request by richarddbarnett**
`winmm.dll` will only work on Windows, not essential to solution.
Change delays because of this - assume 15ms resolution across all
platforms.
Clarify readme.
Conform to our own coding standard.

### CODE REVIEW

**Comment by PeterKnealeNearmap** [comment]
👍

**Review by PeterKnealeNearmap — APPROVED** [review]

File: NMTest.DataSource/DatabaseStore.cs [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
-using System.Linq;
-using System.Text;
+﻿using System.Collections.Generic;
 using System.Threading;
 
 namespace NMTest.DataSource
 {
     public class DatabaseStore
     {
+        private readonly Dictionary<string, object> _values = new Dictionary<string, object>();
```
Comment (jevonkendon) [inline NMTest.DataSource/DatabaseStore.cs:8]: I'm inclined to leave the `private` access modifiers absent, and instead leave those as bonus points for the candidate.

File: NMTest.Sample/Program.cs [hunk 2]
```diff
@@ -1,35 +1,17 @@
...
         {
-            for (int i = 0; i < 10; i++)
+            for (var i = 0; i < 10; i++)
             {
                 new Thread(() =>
                 {
-                    //Sets the resolution of the timer. This has to be done per thread to ensure that 5ms sleep is actually 5 ms.
-                    //http://stackoverflow.com/a/522681
-                    timeBeginPeriod(5);
-                    //Your code should go here.
```
Comment (jevonkendon) [inline NMTest.Sample/Program.cs:28]: I think that's a nice comment to leave in.

File: README.md [hunk 3]
```diff
@@ -4,24 +4,45 @@ Welcome to the [nearmap.com](nearmap.com) C# test. The purpose of this assignmen
...
+* Data in `DatabaseStore` never changes and can be cached forever.
+* If `DatabaseStore.GetValue()` returns `null` for a key, the requested data item does not exist and will never exist.
+* `DistributedCacheStore` is initially empty.
 
 ## Task
+
+Complete the 2 parts below & submit the solution. 
+If the solution is incomplete, please state what hasn't been finished and outline how you are planning on solving it.
+
+* Provided code can be modified at will.
```
Comment (jevonkendon) [inline README.md:27]: "Provided code can be modified at will, including any improvements you wish to make"?

File: README.md [hunk 4]
```diff
@@ -4,24 +4,45 @@ Welcome to the [nearmap.com](nearmap.com) C# test. The purpose of this assignmen
...
+* The whole solution must build with no errors.
```
Comment (jevonkendon) [inline README.md:28]: and warnings?

**Review by jevonkendon — APPROVED** [review]
:rocket:

File: README.md (hunk 4 above)
Comment (richarddbarnett) [reply README.md:28]: ISTR some warnings can't be avoided on Mono

File: NMTest.Sample/Program.cs (hunk 2 above)
Comment (richarddbarnett) [reply NMTest.Sample/Program.cs:28]: I removed it because the `Main()` also needs code before the loop, to populate `DatabaseStore`.

Comment (richarddbarnett) [reply NMTest.Sample/Program.cs:28]: reinstated, & added another.

**Pull request merged**